internal/store/store.go

```diff
@@ -468,6 +468,11 @@ func (s *Store) PushCompose(composeID uuid.UUID, bp *blueprint.Blueprint, checks
 	if err != nil {
 		return err
 	}
+
+	if pipeline.Assembler != nil {
+		size = pipeline.Assembler.Options.GetImageSize()
```

Comment:
Rather than reading this out from the Pipeline, how about introducing a `distro.GetSizeForOutputType(size uint64)` function? This could then also deal with rounding (as is done for vhd in the weldr API), and if you pass it 0 it uses the default value. Then you could store that in the compose, and also pass it back to the pipeline translation.

In general, I'd rather avoid adding any logic to the pipeline package apart from its current role as a thin wrapper around the json objects.